prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -181,12 +183,13 @@ object FHIRBundleHelpers {
         // Go through every resource and check if the resource has a reference to the resource being deleted
         // if there is remove the reference
         allResources.forEach { res ->
-            res.resource.getResourceProperties().forEach { property ->
-                property.values.forEach {
-                    if (it is Reference && it.reference == resource.idBase) {
-                        it.reference = null
-                        it.resource = null
-                    }
+            res.resource::class.memberProperties.forEach { it ->
+                it.isAccessible = true
+                val value = it.getter.call(res.resource)
+                if (value is MutableList<*>) {
+                    value.removeIf { it is Reference && it.reference == resource.idBase }
+                } else if (value is Reference) {
+                    value.reference = null
                 }
```

Comment:
This a fix for a bug found while testing condition filters. The bug happened when we have a bundle with more than two conditions. If a condition was removed in the middle of the list the FHIR to HL7 transform would still try to map that observation. This happened because the linux4health library still sees the entry in the list and tries to map it even though the resource and reference are null. To fix this I had to use reflection to actually remove the entry from the list.

Reply:
Do we have tests for this? We should write tests that catch the current failure and then see that they pass with your fix, right?

Reply:
We would need an integration test to test this. It happens when you remove an observation and then convert that to HL7. We haven't been writing integration tests for UP. Should we create a separate ticket to start doing that? For now it can be tested by following the tests steps and seeing the observations on the generated HL7 message, it wasn't doing that without this change.